### PR TITLE
chore: Bump OpenSSL to 3.0.15

### PR DESCRIPTION
**Pull request by codingllama**
Update to the latest security patch.

- https://github.com/openssl/openssl/blob/openssl-3.0.15/CHANGES.md#openssl-30
- https://openssl-library.org/news/secadv/20240903.txt

Changelog: Updated OpenSSL to 3.0.15

### CODE REVIEW

**Comment by codingllama** [comment]
Tested manually using `tsh fido2 diag`, MFA logins (a few different keys), passwordless registration and passwordless login.

File: build.assets/buildbox/thirdparty-libs.mk [hunk 1]
```diff
@@ -160,9 +160,9 @@ tp-build-libcbor: fetch-git-libcbor
 # -----------------------------------------------------------------------------
 # openssl
 
-openssl_VERSION = 3.0.13
```
Comment (codingllama) [inline build.assets/buildbox/thirdparty-libs.mk:163]: @camscale the versions here were all 1 behind other files. I'm not sure if there was a reason for it or if it was just lagging, but I updated it anyway.

**Review by tigrato — APPROVED** [review]

**Review by rosstimothy — APPROVED** [review]

File: build.assets/buildbox/thirdparty-libs.mk (hunk 1 above)
Comment (camscale) [reply build.assets/buildbox/thirdparty-libs.mk:163]: No worries. I am currently backporting this to the branches, so this won't get backported now but I'll include it in the backports.

Comment (codingllama) [reply build.assets/buildbox/thirdparty-libs.mk:163]: Be careful to match the versions to the branch, as they differ in some way in almost every branch. We backport security fixes like these, but we rarely (if ever) backport increased functionality.

**Pull request merged**